pytensor/mermaid.py

```diff
@@ -0,0 +1,63 @@
+from pytensor.d3viz.formatting import PyDotFormatter
+
+
+def function_to_mermaid(fn):
+    formatter = PyDotFormatter()
+    dot = formatter(fn)
+
+    nodes = dot.get_nodes()
+    edges = dot.get_edges()
+
+    mermaid_lines = ["graph TD"]
+    mermaid_lines.append("%% Nodes:")
+    for node in nodes:
+        name = node.get_name()
+        label = node.get_label()
+        shape = node.get_shape()
+
+        if label.endswith("."):
+            label = f"{label}0"
+
+        if shape == "box":
+            shape = "rect"
+        else:
+            shape = "rounded"
+
+        mermaid_lines.extend(
+            [
+                f'{name}["{label}"]',
+                f"{name}@{{ shape: {shape} }}",
+            ]
+        )
+
+        fillcolor = node.get_fillcolor()
+        if fillcolor is not None and not fillcolor.startswith("#"):
+            fillcolor = _color_to_hex(fillcolor)
+            mermaid_lines.append(f"style {name} fill:{fillcolor}")
+
+    mermaid_lines.append("\n%% Edges:")
+    for edge in edges:
+        source = edge.get_source()
+        target = edge.get_destination()
+
+        mermaid_lines.append(f"{source} --> {target}")
+
+    return "\n".join(mermaid_lines)
+
+
+def _color_to_hex(color_name):
```

Comment:
Why is this a function?

Reply:
mermaid needs hexcolors. the pydotformatter has strings names for colors